### DISCUSSION TITLE
Add deepseek V3  AI

### DISCUSSION THREAD

**Discussion by apptvltd** [cline/cline#1036] [Feature Requests]
They claim to have the best AI benchmarks so for code

https://www.deepseek.com/

**Comment by deiangi** [comment]
as a workaround - I am using the openrouter hosting. its called deepseek/chat - its quite good btw almost like cline but cheaper

**Comment by yan5xu** [comment]
You can use deepseek v3 with the following configuration:

1. API Provider: Select "OpenAI Compatible"
2. Base URL: Enter `https://api.deepseek.com`
3. API Key: Input your deepseek API key
4. Model ID: Use `deepseek-chat`

**Reply by apptvltd** [reply]
Awesome!

**Comment by 0xdileep** [comment]
Hi, could you tell me how to set max tokens to 8k?

**Comment by saoudrizwan** [comment]
Cline now has a new DeepSeek API option in Settings > API Provider. The latest v3.0.12 also fixes issues with DeepSeek's context window limit and trouble with diff editing.

**Comment by synthetic-ape** [comment]
is it possible to run deepseek locally with cline?

**Reply by elibroftw** [reply]
Try [twinny](https://github.com/twinnydotdev/twinny) with [ollama](https://docs.twinny.dev/general/quick-start/#installing-ollama-as-an-inference-provider)

**Comment by Lee-ux-D** [comment]
Hello, I call deepseek API in vs code, it keeps turning but not outputting the content, then it shows "Cannot read properties of undefined (reading 'type'), does anyone know how to solve it?

**Comment by any789** [comment]
If anyone wants a pay-as-you-go option for DeepSeek V4-Flash, Cheap AI API (topaitools.vip/token-chuhai.html) works at $1/$6 per 1M tokens. OpenAI-compatible, so you just swap the base URL. Their bot @CheapAIApiBot gives you the key right after payment.
